processor/src/decorators/mod.rs

```diff
@@ -1,14 +1,21 @@
-use super::{AdviceInjector, ExecutionError, Felt, Process, StarkField};
+use super::{AdviceInjector, Decorator, ExecutionError, Felt, Process, StarkField};
 
 // DECORATORS
 // ================================================================================================
 
 impl Process {
+    /// Executes the specified decorator
+    pub(super) fn execute_decorator(&mut self, decorator: Decorator) -> Result<(), ExecutionError> {
+        match decorator {
+            Decorator::Advice(injector) => self.decorator_advice(injector)?,
+        }
+        Ok(())
+    }
     // ADVICE INJECTION
     // --------------------------------------------------------------------------------------------
 
     /// Process the specified advice injector.
-    pub fn op_advice(&mut self, injector: AdviceInjector) -> Result<(), ExecutionError> {
+    pub fn decorator_advice(&mut self, injector: AdviceInjector) -> Result<(), ExecutionError> {
```

Comment:
nit: should we rename this to `dec_advice`?

Reply:
Initially I considered that name but it didn't seem really clear and `decorator_advice` didn't seem long enough to necessitate abbreviation. Will defer to your choice.

Reply:
I have changed it to dec_advice for now.